etl/steps/data/meadow/usgs/2024-07-15/historical_statistics_for_mineral_and_material_commodities.py

```diff
@@ -0,0 +1,174 @@
+"""Load a snapshot and create a meadow dataset."""
+
+import re
+import tempfile
+import warnings
+from pathlib import Path
+from zipfile import ZipFile
+
+import owid.catalog.processing as pr
+import pandas as pd
+
+from etl.helpers import PathFinder, create_dataset
+from etl.snapshot import Snapshot
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Ignore UserWarnings from openpyxl, that repeatedly shows "Unknown extension is not supported and will be removed", even though the loading works well.
+warnings.filterwarnings("ignore", category=UserWarning, module="openpyxl")
+
+
+def read_data_for_all_commodities(snap: Snapshot):
+    # Initialize a dictionary that will gather the excel supply-demand data for each file.
+    supply_demand_data = {}
+    # Open the zip file.
+    with ZipFile(snap.path, "r") as zipf:
+        # Create a temporary directory.
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_path = Path(temp_dir)
+
+            # Extract supply_demand files.
+            supply_demand_dir = temp_path / "supply_demand"
+            zipf.extractall(path=temp_path)
+
+            # Iterate through the extracted files and apply process_file().
+            for file_path in supply_demand_dir.glob("*.xlsx"):
+                supply_demand_data[file_path.stem] = pr.ExcelFile(file_path)
```

Comment:
Good idea. Ideally, we should assert that a list of expected files are in the zip, but also (optionally) warn if new files are added (because, if new minerals are included, I'd like to fetch them too). Feel free to implement it yourself or create an issue (I don't have the appetite for this PR).